modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -607,18 +607,35 @@ object EnrichmentManager {
     }
 
   def getCrossDomain[F[_]: Applicative](
-    pageQsMap: Option[QueryStringParameters]
+    pageQsMap: Option[QueryStringParameters],
+    crossNavEnrichment: Option[CNE]
   ): EStateT[F, Unit] =
     EStateT.fromEither {
       case (event, _) =>
         pageQsMap match {
           case Some(qsMap) =>
-            val crossDomainParseResult = WPE.parseCrossDomain(qsMap)
-            for ((maybeRefrDomainUserid, maybeRefrDvceTstamp) <- crossDomainParseResult.toOption) {
-              maybeRefrDomainUserid.foreach(event.refr_domain_userid = _)
-              maybeRefrDvceTstamp.foreach(event.refr_dvce_tstamp = _)
-            }
-            crossDomainParseResult.bimap(NonEmptyList.one(_), _ => Nil)
+            WPE
+              .parseCrossDomain(qsMap)
+              .bimap(
+                err =>
+                  crossNavEnrichment match {
+                    case Some(cn) => NonEmptyList.one(cn.addEnrichmentInfo(err))
+                    case None => NonEmptyList.one(err)
+                  },
+                crossNavMap => {
+                  val crossNavKeys = crossNavMap.keySet
+                  val duidKey = CNE.CrossNavProps(0)
```

Comment:
This seems a bit unsafe and hard to follow as it is. I think it would better if we refactor out `duidKey` to some variable in `CrossNavigationEnrichment` and use that instead.

Or, we can solve this problem with wrapping `crossNavMap` to some class such as `CrossDomainMap`. With this way, we wouldn't have to bother witn map itself or its key names. It can be abstracted out to a method in `CrossDomainMap` class.

I've implemented what I had in mind in [here](https://github.com/snowplow/enrich/tree/feat/cross_navigation_cleanup).